sonar-ballerina-plugin/build.gradle

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.jar.JarInputStream
+
+plugins {
+    id 'java'
+    id 'java-library'
+    id 'checkstyle'
+    id "com.github.spotbugs"
+    id "de.undercouch.download"
+    id "com.github.johnrengelman.shadow"
+    id "io.spring.dependency-management"
+}
+
+group = "${group}"
+version = "${sonarPluginVersion}"
+
+repositories {
+    mavenLocal()
+    mavenCentral()
+    gradlePluginPortal()
+
+    maven {
+        url = 'https://repox.jfrog.io/repox/plugins.gradle.org/'
+    }
+
+    maven {
+        url = uri("https://plugins.gradle.org/m2/")
+    }
+}
+
+dependencies {
+    compileOnly group: 'org.sonarsource.api.plugin', name: 'sonar-plugin-api', version: "${sonarPluginAPIVersion}"
+
+    implementation group:'javax.annotation', name:'javax.annotation-api', version: "${javaAnnotationAPIVersion}"
+    implementation group: 'org.sonarsource.analyzer-commons', name: 'sonar-analyzer-commons', version: "${sonarAnalyzerCommonsVersion}"
+    implementation group: 'org.sonarsource.analyzer-commons', name: 'sonar-xml-parsing', version: "${sonarAnalyzerCommonsVersion}"
+    implementation group: 'org.sonarsource.analyzer-commons', name: 'sonar-regex-parsing', version: "${sonarAnalyzerCommonsVersion}"
+    implementation group: 'org.sonarsource.analyzer-commons', name: 'sonar-performance-measure', version: "${sonarAnalyzerCommonsVersion}"
+    implementation group: 'org.sonarsource.analyzer-commons', name: 'sonar-analyzer-recognizers', version: "${sonarAnalyzerCommonsVersion}"
+    implementation group: 'org.slf4j', name: 'slf4j-api', version: "${slf4jVersion}"
+    implementation group: 'ch.qos.logback', name: 'logback-classic', version: "${logbackVersion}"
```

Comment:
@Xenowa 
Is there any reason to use this? We need to review this License. If possible better to use slf4j/log4j

Reply:
These dependencies are required when writing plugin tests for `SonarQube v9.9 LTS` when used with the `sonar-plugin-api V9.14.0.375`. The plugin API has a dependency on the logback which requires the `slf4j` dependencies to be included in Gradle to prevent `ClassCastExceptions` that occur as follows during testing:

```bash
class org.slf4j.helpers.NOPLogger cannot be cast to class ch.qos.logback.classic.Logger (org.slf4j.helpers.NOPLogger and ch.qos.logback.classic.Logger are in unnamed module of loader 'app')
java.lang.ClassCastException: class org.slf4j.helpers.NOPLogger cannot be cast to class ch.qos.logback.classic.Logger (org.slf4j.helpers.NOPLogger and ch.qos.logback.classic.Logger are in unnamed module of loader 'app')
	at org.sonar.api.utils.log.LogbackLoggers.newInstance(LogbackLoggers.java:34)
	at org.sonar.api.utils.log.LoggerFactory.get(LoggerFactory.java:48)
	at org.sonar.api.utils.log.Loggers.get(Loggers.java:35)
	at org.sonar.api.batch.fs.internal.AbstractProjectOrModule.<clinit>(AbstractProjectOrModule.java:42)
	at org.sonar.api.batch.sensor.internal.SensorContextTester.<init>(SensorContextTester.java:126)
	at org.sonar.api.batch.sensor.internal.SensorContextTester.create(SensorContextTester.java:136)
	at io.ballerina.sonar.AbstractSensorTest.setup(AbstractSensorTest.java:43)
```

These dependencies have been removed in the `sonar-plugin-api V9.15.0.435` as provided in the [change log](https://github.com/SonarSource/sonar-plugin-api/releases/tag/9.15.0.435), Therefore they can be removed after bumping up the `sonar-plugin-api` version without any issues.